core/src/main/java/org/apache/carbondata/core/scan/filter/FilterUtil.java

```diff
@@ -188,6 +189,14 @@ private static FilterExecuter createFilterExecuterTree(
           return new FalseFilterExecutor();
         case ROWLEVEL:
         default:
+          if (filterExpressionResolverTree.getFilterExpression() instanceof UnknownExpression) {
+            FilterExecuter filterExecuter =
+                ((UnknownExpression) filterExpressionResolverTree.getFilterExpression())
+                    .getFilterExecuter(filterExpressionResolverTree, segmentProperties);
+            if (filterExecuter != null) {
```

Comment:
If in case, filterExecuter is null, we need to return RowLevelFilterExecuterImpl right?. If filterExecuter never comes null, can check and remove this check

Reply:
I understand what you meant. I think, it is better to create the RowLevelFilterExecuterImpl only at this place. If UnknownExpression do not have any specific implemenation, it will be like a fallback to RowLevelFilterExecuterImpl.
 Also, if we have return RowLevelFilterExecuterImpl from UnknownExpression.getFilterExecuter, we need to pass complexDimensionInfoMap and all overiding methods will have this argument. It is an unused argument as current UnKnown expressions do not support complex dimensions.